src/features/dashboard/components/PieChartCard.tsx

```diff
@@ -26,14 +26,17 @@ export type DiversificationType =
 export const PieChartCard = () => {
   const [selectedType, setSelectedType] =
     useState<DiversificationType>('Aktiva');
-  const { holdingData, sectorData, typeData, isLoading } = usePieChartsData();
+  const { holdingData, sectorData, typeData, dividendData, isLoading } =
+    usePieChartsData();
 
   const chartData =
     selectedType === 'Sektor'
       ? sectorData
       : selectedType === 'Typ aktiva'
         ? typeData
-        : holdingData;
+        : selectedType === 'Aktiva'
+          ? holdingData
+          : dividendData;
```

Comment:
This conditional assigning can get out of hand really quick. Yous should use a mapping object instead
```ts
 const chartDataMap = {
    Sektor: sectorData,
    'Typ aktiva': typeData,
    Aktiva: holdingData,
    Dividendy: dividendData,
  };

  const chartData = chartDataMap[selectedType];
```